patches/minecraft/net/minecraft/world/gen/layer/OceanLayer.java.patch

```diff
@@ -0,0 +1,25 @@
+--- a/net/minecraft/world/gen/layer/OceanLayer.java
++++ b/net/minecraft/world/gen/layer/OceanLayer.java
+@@ -11,13 +11,18 @@
+       ImprovedNoiseGenerator improvednoisegenerator = p_215735_1_.func_205589_a();
+       double d0 = improvednoisegenerator.func_215456_a((double)p_215735_2_ / 8.0D, (double)p_215735_3_ / 8.0D, 0.0D, 0.0D, 0.0D);
+       if (d0 > 0.4D) {
+-         return LayerUtil.field_203632_a;
++         return net.minecraft.util.registry.Registry.field_212624_m.func_148757_b(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.OceanType.WARM, p_215735_1_).biome);
+       } else if (d0 > 0.2D) {
+-         return LayerUtil.field_203633_b;
++         return net.minecraft.util.registry.Registry.field_212624_m.func_148757_b(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.OceanType.LUKEWARM, p_215735_1_).biome);
+       } else if (d0 < -0.4D) {
+-         return LayerUtil.field_202831_b;
++         return net.minecraft.util.registry.Registry.field_212624_m.func_148757_b(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.OceanType.FROZEN, p_215735_1_).biome);
+       } else {
+-         return d0 < -0.2D ? LayerUtil.field_203634_d : LayerUtil.field_202832_c;
++         return d0 < -0.2D ? net.minecraft.util.registry.Registry.field_212624_m.func_148757_b(getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.OceanType.COLD, p_215735_1_).biome) : net.minecraft.util.registry.Registry.field_212624_m.func_148757_b(this.getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.OceanType.NORMAL, p_215735_1_).biome);
+       }
+    }
++
++   public static net.minecraftforge.common.BiomeManager.OceanBiomeEntry getWeightedBiomeEntry(net.minecraftforge.common.BiomeManager.OceanType type, INoiseRandom context) {
```

Comment:
Wouldn't it be better to have a smaller patch by moving this method to an impl class? IIRC forge prefers their patch files to be nice and small so they can fix patches easier across updates.

Reply:
Or is there a reasoning for having the method here?

Reply:
The general consensus is to not override in patches if the base class can define the logic.
To be fair I had to ask as well.

But since this is at the bottom of the patch file it does not matter.

Reply:
I noticed `BiomeLayer` was doing a very similar thing to what I needed for `OceanLayer` at the bottom of the class, so I thought it'd be ok.

So, what do you guys think, keep it in the class or move it to `BiomeManager`?